docs/api/createAsyncThunk.md

```diff
@@ -340,25 +340,29 @@ const usersSlice = createSlice({
   initialState: {
     entities: [],
     loading: 'idle',
+    currentRequestId: undefined,
     error: null
   },
   reducers: {},
   extraReducers: {
     [fetchUserById.pending]: (state, action) => {
       if (state.loading === 'idle') {
         state.loading = 'pending'
+        state.currentRequestId = action.meta.requestId
       }
     },
     [fetchUserById.fulfilled]: (state, action) => {
       if (state.loading === 'pending') {
```

Comment:
Doing a `(state.currentRequestId === action.meta.requestId)` test here is much more important than looking for the correct loading state.

Even if the asyncThunk exits early, it will dispatch a pending action and a rejected or resolved action. So you'd run into this:


* state idle, currentRequestId undefined, entities []
* users/fetchByIdStatus/pending (requestId 1)
* state pending, currentRequestId 1
* users/fetchByIdStatus/pending (requestId 2)
* action is ignored
* payloadCreator for request 2 exits early
* users/fetchByIdStatus/resolved (requestId 2)
* state idle, currentRequestId undefined, entities [undefined]
* users/fetchByIdStatus/resolved (requestId 1)
* action is ignored